Formula/killswitch.rb

```diff
@@ -0,0 +1,16 @@
+class Killswitch < Formula
+  desc "VPN kill switch for macOS"
+  homepage "https://vpn-kill-switch.com"
+  url "https://github.com/vpn-kill-switch/killswitch/archive/0.7.1.tar.gz"
+  sha256 "64be2ea4ebc62cea724b3dad7ddf61a6be7ea285012f99651380f7b189aa0dc5"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", "-mod=readonly", "-ldflags", "-s -w -X main.version=#{version}", "-o", "#{bin}/killswitch", "cmd/killswitch/main.go"
+  end
+
+  test do
+    system "#{bin}/killswitch", "-v"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Hi @SMillerDev the problem, in this case and mainly because of the nature of the application, is difficult to come up with a "test case", for example, to create a file with the firewall rules, the system running the test should have a VPN connection up and running, and in case it had it, the VPN peer IP may be different, therefore every time running the tests the "output" would be different.

The code has some tests `go test -v` don't know if that could be helpful for this case?

Reply:
Hi @SMillerDev I updated the formula testing that it fails and matches the stderr output, please take a look.